File: 0009.PalindromeNumber.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        #return str(x) == str(x)[::-1]
        if x<0 or (x!=0 and x%10==0):
            return False
        y = 0
        while x > y:
            y = 10*y + x%10
            x /= 10
        return y == x or y/10 == x
        

File: 0009.PalindromeNumber.go
func isPalindrome(x int) bool {
    if x < 0 || (x != 0 && x%10 == 0) {
        return false
    }
    y := 0
    for y < x {
        y = 10*y + x%10
        x /= 10
    }
    return y == x || y/10 == x
    /*
    if x < 0 {
        return false
    }
    nums := []int{}
    for x > 0 {
        nums = append(nums, x%10)
        x /= 10
    }
    for i, j := 0, len(nums)-1; i < j; i, j = i+1, j-1 {
        if nums[i] != nums[j] {
            return false
        }
    }
    return true
    */
}